Doc/library/typing.rst

```diff
@@ -849,6 +849,22 @@ using ``[]``.
       concat(b"foo", b"bar")  # OK, output has type 'bytes'
       concat("foo", b"bar")   # Error, cannot mix str and bytes
 
+   Note that ``AnyStr`` and ``str | bytes`` are different from each other and
+   have different use cases. ``str | bytes`` does allow you to mix str and bytes::
+
+      def concat(a: str | bytes, b: str | bytes) -> str | bytes:
+          return a + b
+
+      concat("foo", b"bar")  # Passes typecheck but raises an error at runtime
+
+   As a type variable, ``AnyStr`` is only used if it's at least part of the input arguments::
```

Comment:
This sentence makes it sound like you are introducing an example where AnyStr is used correctly, but in fact you're introducing an incorrect usage.